### PR TITLE
Update Watchlist Class and its testing methods

### PR DESCRIPTION
**Pull request by OngDeZhi**
Update: 
1. Watchlist class's equals() method.
2. Testing methods for Watchlist class.

### CODE REVIEW

**Review by ChanJianHao — APPROVED** [review]
Great work, lgtm.

**Review by michaeldinata — APPROVED** [review]
LGTM! :)

File: src/test/java/seedu/duke/watchlist/WatchlistTest.java [hunk 1]
```diff
@@ -2,66 +2,39 @@
...
-    private Watchlist emptyWatchlist;
-    private Watchlist filledWatchlist;
+    private Watchlist firstWatchlist;
+    private Watchlist secondWatchlist;
 
     @BeforeEach
     void setUp() {
         ArrayList<Integer> animeList = new ArrayList<>();
-        animeList.add(0);
+        animeList.add(1);
```
Comment (ChanJianHao) [inline src/test/java/seedu/duke/watchlist/WatchlistTest.java]: Perhaps use of a constant would provide more clarity.

Comment (OngDeZhi) [reply src/test/java/seedu/duke/watchlist/WatchlistTest.java]: Noted on that, I have updated to use a constant for it.

**Pull request merged**